htsohm/htsohm.py

```diff
@@ -43,61 +37,64 @@ def queue_all_materials(run_id, generation, queue):
+    print("== manhpc_job_dummy_test %s" % material_id)
+
+    parent_id = screen_parent(run_id)
+    write_child_definition_files(run_id, generation, parent_id)
+    session.commit()
+    print("======================================================================================")
 
-def seed_generation(run_id, children_per_generation, number_of_atomtypes, queue=None):
+def queue_create_next_gen(run_id, generation, queue):
+    """same as create_all_materials, except queues the jobs in the job server"""
+    children_per_generation = read_config_file(run_id)['children-per-generation']
+    for i in range(children_per_generation):
+        trials = read_config_file(run_id)['dummy-test-trials']
+        queue.enqueue(hpc_job_create_material, run_id, generation, timeout=trials*60*60)
+
+def seed_generation(run_id, children_per_generation, number_of_atomtypes):
     generation = 0
-    init_materials_in_database(run_id, children_per_generation, generation)
     write_seed_definition_files(run_id, children_per_generation, number_of_atomtypes)
-    if queue is not None:
-        queue_all_materials(run_id, generation, queue)
-    else:
-        simulate_all_materials(run_id, generation)
+    session.commit()
 
-def next_generation(run_id, children_per_generation, generation, queue=None):
-    init_materials_in_database(run_id, children_per_generation, generation)
-    screen_parents(run_id, children_per_generation, generation)
-    create_next_generation(run_id, generation)
-    if queue is not None:
-        queue_all_materials(run_id, generation, queue)
-    else:
-        simulate_all_materials(run_id, generation)
+def update_strength_array(run_id, generation):
+    if generation == 1:
+        create_strength_array(run_id)
+    elif generation >= 2:
+        recalculate_strength_array(run_id, generation)
 
 def htsohm(children_per_generation,    # number of materials per generation
            number_of_atomtypes,        # number of atom-types per material
            strength_0,                 # intial strength parameter
            number_of_bins,             # number of bins for analysis
            max_generations=20,         # maximum number of generations
-           acceptance_value=-0.5):      # desired degree of `convergence`
+           dummy_test_trials=1,        # number of re-simulations for dummy-test
+           acceptance_value=-0.005):      # desired degree of `convergence`
     ############################################################################
     # write run-configuration file
     run_id = write_config_file(children_per_generation, number_of_atomtypes, strength_0,
-        number_of_bins, max_generations)["run-id"]
+        number_of_bins, max_generations, dummy_test_trials, acceptance_value)["run-id"]
 
     convergence = acceptance_value + 1          # initialize convergence with arbitrary value
     for generation in range(max_generations):
         while convergence >= acceptance_value:
```

Comment:
I think I've covered everything we went over!

a

On Tue, Jul 5, 2016 at 12:13 PM, Paul Boone <notifications@github.com>
wrote:

> In htsohm/htsohm.py
> <https://github.com/WilmerLab/HTSOHM-dev/pull/8#discussion_r69592270>:
>
> >      ############################################################################
> >      # write run-configuration file
> >      run_id = write_config_file(children_per_generation, number_of_atomtypes, strength_0,
> > -        number_of_bins, max_generations)["run-id"]
> > +        number_of_bins, max_generations, dummy_test_trials, acceptance_value)["run-id"]
> >
> >      convergence = acceptance_value + 1          # initialize convergence with arbitrary value
> >      for generation in range(max_generations):
> >          while convergence >= acceptance_value:
>
> the for loop here isn't doing anything any more since the while loop will
> continue until completed
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/WilmerLab/HTSOHM-dev/pull/8/files/826958b0372cabb991caeb6e77907a1e09c7223e#r69592270>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHtyl_o-M9lS3fuUP6IDD-b0u5jb_yfaks5qSoKkgaJpZM4JDbXv>
> .
>